src/applications/edu-benefits/containers/PreSubmitInfo.jsx

```diff
@@ -0,0 +1,78 @@
+import React, { useEffect } from 'react';
+import Checkbox from '@department-of-veterans-affairs/component-library/Checkbox';
+import { connect } from 'react-redux';
+
+// platform - form-system actions
+import { setPreSubmit as setPreSubmitAction } from 'platform/forms-system/src/js/actions';
+
+function PreSubmitNotice({
+  formData,
+  showError,
+  onSectionComplete,
+  setPreSubmit,
+}) {
+  const privacyAgreementAccepted = formData.privacyAgreementAccepted || false;
+  // const vrrapConfirmation = formData.vrrapConfirmation;
+
+  // set AGREED (onSectionComplete) to value of privacyAgreementAccepted
+  // if goes to another page (unmount), set AGREED (onSectionComplete) to false
+  useEffect(
+    () => {
+      onSectionComplete(privacyAgreementAccepted);
+
+      return () => {
+        onSectionComplete(false);
+      };
+    },
+
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```

Comment:
Is there a reason you are needing to comment out this linting rule? Seems like one we wouldn't want to ignore...

Reply:
Hi @k80bowman, it looks like the reason that is disabled is because `useEffect` is expecting `onSectionComplete` in the dependency array. This was copied over from: https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/applications/caregivers/components/PreSubmitInfo/index.jsx#L74 which was used as an example to set up the `PreSubmitInfo` component for the 1990s form. With `onSectionComplete` added to that array or with the array removed entirely, the form "hangs up" and cannot receive input or be submitted which appears to be why it is disabled.

Reply:
Thanks for the response @elder-joseph. In this case in particular, I'm concerned that not finding a way to implement this feature while following the Rules of Hooks (what this rule is trying to enforce) could cause serious problems with state management, performance, and could impart flakiness to our tests. I don't know enough about your code here (or how to run it locally) to suggest a fix, but I would appreciate it if you could take a look and see if there is a way to fix this.

One good thing to check first would be if this effect is happening on every re-render of this component (which could result in a performance hit). You could check this with a few simple unit tests such as:

> // render 
// expect(effect).to.have.been.called.once
// fire event that doesn't change effect dependency
// expect(effect).to.have.been.called.once
// fire event that does change effect dependency
// expect(effect).to.have.been.called.twice

Reply:
Hi @k80bowman, I agree its usually not a good idea to disable linter warnings as they are there for a reason. After digging in a bit, it looks as if adding the `onSectionComplete` to the dependency array for `useEffect` was causing an infinite loop when rendering the component. Passing an empty array was a solution to this, however, not ideal since we would be omitting both dependencies from the array. The solution I found was to replace `useEffect` with `useCallback`, that way the function is only executed when needed. I pushed an update to use this in the latest commit.

Reply:
@elder-joseph thank you so much for digging into this. I'm glad you were able to find a fix. I would still recommend adding some tests, just to be sure it's working the way you want it to, but this looks much better.

Reply:
Hi @k80bowman, no problem. I am working on cleaning up a unit test for this right now and will push it once finished. I appreciate your insight here.

Reply:
👋 Just jumping in to point out that the `useCallback` is typically used to memoize a function that you pass down to subcomponents, so I'm not sure that it's being used correctly in this change. I believe that the parent component of `PreSubmitInfo` is supposed to be setting a memoized version of `onSectionComplete` as a prop in `PreSubmitInfo`. I don't have enough context into the structure of your app, but I'd expect something like this:

```
function PreSubmitInfoParent({ ... }) {
  ...

  const memoizedOnSectionComplete = useCallback(..., [onSectionComplete]);

  return (
    <PreSubmitInfo onSectionComplete={memoizedOnSectionComplete}>
      ...
    </PreSubmitInfo>
  );
}

function PreSubmitInfo({ ... }) {
  ...

  useEffect(..., [onSectionComplete, privacyAgreement]);

  ...
}
```

I suspect the reason you got an infinite loop previously by simply passing in `onSectionComplete` in the `useEffect` dependency array is because the parent component wasn't using `useCallback` to mark that `onSectionComplete` didn't change and didn't need to trigger a re-render in `PreSubmitInfo`.

Reply:
Hi @U-DON I appreciate the feedback. I agree with your assessment as to the reason why I was seeing an infinite loop with the dependency being passed. 

The parent component of `PreSubmitInfo` is https://github.com/department-of-veterans-affairs/vets-website/blob/master/src/platform/forms/components/review/PreSubmitSection.jsx which is used by other forms outside of our team's scope. 

I am hesitant to modify that file and potentially cause issues with submission of those other forms. I'm not entirely sure what the best resolution to this would be.

Reply:
I realized shortly after that the change would involve Platform code. Since it's in the Forms Library, I've left it to the Design System team (@cvalarida) to determine whether this is something they want to fix in the short term. If not, I believe the original `useEffect` with the disabled linting rule would be closer to the behavior that you intended.

Reply:
@U-DON thanks for taking a look. After trying some different ideas here, I found I was able to remove that entire section of code related to this discussion altogether. I pushed this change in the latest commit as it appears to have the behavior we expect. If for whatever reason this does not pass our local validation, and we are unable to change the platform component, I will revert back to the `useEffect` with the disabled linting rule.